feat: Automatic Job Deletion

vargasman11 opened the pull request:
- Creates an entries controller action for deletion of closed jobs older than 6 months
- Can be configured via config file

ConversioniaDev commented on src/console/controllers/EntriesController.php:
```diff
@@ -0,0 +1,90 @@
+        $settings = Leadflex::$plugin->getSettings();
+
+        // Check to see if action is enabled
+        $isJobDeletionEnabled = $settings->isJobDeletionEnabled;
```
Can you make the config / property more alike with the action name?
DeleteStaleJobs -> isDeleteStaleJobsEnabled.

ConversioniaDev commented on src/config.php:
```diff
@@ -39,5 +39,11 @@
+    // Enable automatic deletion of stale job entries
+    "isJobDeletionEnabled" => false,
+
+    // Months disabled jobs should are to be kept before deletion - DateTime::modify(), such as: '6 months' (valid format)
```
The values passed could be any Day-based notation. A config could pass "1 year", which would then separate the variable name with the value.  

- Make the variable name more aligned / vague with the passable values.
- Enforce only months.

ConversioniaDev commented on src/console/controllers/EntriesController.php:
```diff
@@ -0,0 +1,90 @@
+        $numberOfMonths = $settings->jobDeletionMonths;
+
+        $today = new \DateTime();
+        $monthThreshold= $today->modify("-${numberOfMonths}")->format('Y-m-d');
```
However you choose to control the duration before deleting a stale job - variable name to match.

ConversioniaDev commented on src/console/controllers/EntriesController.php:
```diff
@@ -0,0 +1,90 @@
+            ->all();
+
+        // Check if any job entries are found
+        if (empty($entries)) {
```
Good logging

ConversioniaDev commented on src/console/controllers/EntriesController.php:
```diff
@@ -0,0 +1,90 @@
+                echo "Entry successfully deleted: " . $entry->id . "\n";
+                $deletedCount++;
+            } else {
+                Craft::error("Failed to delete entry: " . $entry->id . "\n");
```
Good use of Craft:info and Craft::error

ConversioniaDev reviewed (changes requested):
Logic is solid. Good logging with it as well. Just a couple 'readability' feedback items.

vargasman11 commented:
Hello @JeffreyBenusa,

I've pushed up the changes you have mentioned.

 - Renamed the configuration variable from `jobDeletionMonths` to `jobDeletionDuration` 
 - updated doc blocks regarding time unit durations

JeffreyBenusa reviewed (approved):
Looks good.
